jdaviz/configs/imviz/plugins/coords_info/coords_info.py

```diff
@@ -489,42 +490,42 @@ def _image_viewer_update(self, viewer, x, y):
+                # 'unknown' rather than surface brightness, so multiply out pix2
+                # and check if the numerator is a spectral/photon flux density
                 if check_if_unit_is_per_solid_angle(unit, return_unit=True) == PIX2:
-                    un = u.Unit(unit) * PIX2
-                    physical_type = un.physical_type
+                    physical_type = (u.Unit(unit) * PIX2).physical_type
                 else:
                     physical_type = u.Unit(unit).physical_type
 
-                if str(physical_type) not in ("spectral flux density",
-                                              "surface brightness"):
-                    skip_spectral_density_eqv = True
+                valid_physical_types = ["spectral flux density",
+                                        "surface brightness",
+                                        "surface brightness wav",
+                                        "photon surface brightness wav",
+                                        "photon surface brightness",
+                                        "power density/spectral flux density wav",
+                                        "photon flux density wav",
+                                        "photon flux density"]
 
-                if self.image_unit is not None and not skip_spectral_density_eqv:
-                    if 'PIXAR_SR' in self.app.data_collection[0].meta:
-                        # Need current slice value and associated unit to use to compute
-                        # spectral density equivalencies that enable Flux to Flux conversions.
-                        # This is needed for units that are not directly convertible/translatable.
-                        slice = viewer.slice_value * u.Unit(self.app._get_display_unit('spectral'))
+                if str(physical_type) in valid_physical_types and self.image_unit is not None:
 
-                        value = flux_conversion(value, unit, self.image_unit,
-                                                eqv=_eqv_pixar_sr(self.app.data_collection[0].meta['PIXAR_SR']),  # noqa: E501
-                                                slice=slice)
-                        unit = self.image_unit
+                    # Create list of potentially needed equivalencies for flux/sb unit conversions
+                    pixar_sr = self.app.data_collection[0].meta.get('PIXAR_SR', 1)
+                    cube_wave = viewer.slice_value * u.Unit(self.app._get_display_unit('spectral'))
 
-                    elif self.image_unit.is_equivalent(unit):
-                        value = (value * u.Unit(unit)).to_value(u.Unit(self.image_unit))
-                        unit = self.image_unit
+                    equivalencies = all_flux_unit_conversion_equivs(pixar_sr,
+                                                                    cube_wave)
+
+                    value = flux_conversion_general(value, u.Unit(unit), u.Unit(self.image_unit),
```

Comment:
Since `u.Unit(unit)` is used multiple times in this same method, maybe it should be assigned to a variable first.